### PR TITLE
prepare automated release process via lerna/conventional-commits

### PR DESCRIPTION
**Pull request by basti1302**

### CODE REVIEW

File: CONTRIBUTING.md
Comment (basti1302) [inline CONTRIBUTING.md:119]: I would like to do this in two steps: Let's publish the next 2 or 3 releases still manually and if we encounter no issues there, we can switch to having it done automatically via the Github action (see below). Thus, for now, only the instructions on how to do a release are updated.

File: bin/needs-release.js
Comment (basti1302) [inline bin/needs-release.js:7]: This file is used in the Github action to determine if a new release should be published. Otherwise we might publish releases that actually contain no real changes at all (only build changes etc.)

File: package.json
Comment (basti1302) [inline package.json:38]: This change (linting additional files and directories) is unrelated to the main intent of this PR.

File: package.json
Comment (basti1302) [inline package.json:106]: I have moved grpc to optional dependencies simply because it takes ages to install (lot's of C++ code that needs to be compiled), simply to speed up the `npm install` step in the Github action (via `--no-optional`). It is quite a hack. Let me know if y'all think this is a bad idea. :-)

File: package.json
Comment (basti1302) [inline package.json:36]: this is for `npm run commit`, which brings up an interactive CLI wizard to compose a commit message

File: .github/workflows/release.yaml
Comment (kirrg001) [inline .github/workflows/release.yaml:7]: **For tomorrow's meeting:**
We have the option to either let the Github action determine if a new release should happen (dangerous?) or we keep the decision within the team by merging develop branch and into main which will trigger a full release automatically.

Comment (basti1302) [reply .github/workflows/release.yaml:7]: Yep, that's one of the items on my list 👍 :-)

Comment (basti1302) [reply .github/workflows/release.yaml:7]: Outcome: For now we want to keep a manual trigger for actually publishing a release. We still need to decide if we want to realize that via two branches (development vs. main) or by manually triggering a job to publish a release. IMO we can leave that undecided for now, as this PR (and the GH action) will not actually release anything.

**Comment by basti1302** [comment]
This is ready for review now I think. I'll fix the commits on main (or our commitlint config) today to make the commit messages and our commitlint rules agree with each other.

File: package.json
Comment (kirrg001) [reply package.json:106]: I think it's smart, but IMO I'd not use this, because it's really hard to understand why we are doing this + it's a misusage.

Two other options/ideas:
1. Try https://github.com/actions/cache if the installation takes too long
2. Switch to https://www.npmjs.com/package/@grpc/grpc-js

File: package.json
Comment (kirrg001) [inline package.json:144]: _Suggestion_: Version updates not part of this PR (or as separate commit, which can be merged as a single change)

File: package.json
Comment (kirrg001) [reply package.json:38]: Do you want to move this out of the PR?

**Review by kirrg001 — COMMENTED** [review]
The PR still has a couple of todo's and I think there are some small changes needed before we can merge it e.g. comment out "- run: bin/needs-release.js" for now? 

I'll wait till the PR is in a final shape and review it again then :)

File: package.json
Comment (basti1302) [reply package.json:106]: Option (2.) does not work for us -- we instrument the "old" grpc package, thus we need that package in our dev dependencies (for the integration test). Instrumenting the "new" package `@grpc/grpc-js` is on the list I think, but not done yet, plus it would not help to get rid of the old dependency.

I'll look into (1.) 👍

File: package.json
Comment (basti1302) [reply package.json:144]: Ah, good catch, I wanted to drop that commit again, but forgot. I only did this because the build failed at the `npm audit` stage because of the the ansi-regex vulnerability and didn't even get to the commitlint step on CircleCI. I'll definitely remove that again.

File: package.json
Comment (basti1302) [reply package.json:38]: I would just keep it in this one as it is a really small-ish change unless you feel strongly about having it in a separate PR. The resulting commit history after merging this PR will be the same. (It is a separate commit already.)

File: package.json
Comment (basti1302) [reply package.json:106]: For now I have removed that commit, so grpc is back in the normal devDependencies.

File: package.json
Comment (basti1302) [reply package.json]: I have dropped that commit.

**Comment by basti1302** [comment]
> The PR still has a couple of todo's and I think there are some small changes needed before we can merge it e.g. comment out "- run: bin/needs-release.js" for now?
> 
> I'll wait till the PR is in a final shape and review it again then :)

I have removed bin/needs-release.js again, good call. The remaining TODOs in the Github action are actually meant to stay there for now. But we can also throw out the whole Github action. It was just meant to give us a convenient preview on how `lerna publish` / `lerna version` will behave once we move this to CI.

File: package.json
Comment (kirrg001) [reply package.json:38]: 👍 Works for me

File: commitlint.config.js
Comment (kirrg001) [inline commitlint.config.js:16]: <3

**Review by kirrg001 — APPROVED** [review]
LGTM!

**Pull request merged**
